core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -89,19 +88,37 @@ class TransactionStateManager(brokerId: Int,
   /**
    * Get the transaction metadata associated with the given transactional id, or null if not found
    */
-  def getTransactionState(transactionalId: String): Option[TransactionMetadata] = {
-    Option(transactionMetadataCache.get(transactionalId))
+  def getTransactionState(transactionalId: String): Option[CoordinatorEpochAndTxnMetadata] = {
+    val partitionId = partitionFor(transactionalId)
+
+    transactionMetadataCache.get(partitionId) match {
+      case Some(txnMetadataCacheEntry) =>
+        txnMetadataCacheEntry.metadataPerTransactionalId.get(transactionalId) match {
+          case null => None
+          case txnMetadata => Some(CoordinatorEpochAndTxnMetadata(txnMetadataCacheEntry.coordinatorEpoch, txnMetadata))
+        }
+      case None => None
```

Comment:
ack. this needs to be tweaked a bit since the inner map is a `Pool`, let me know if you like the new pattern or not.